activerecord/lib/active_record/tasks/database_tasks.rb

```diff
@@ -178,7 +178,11 @@ def for_each(databases)
         return if database_configs.count == 1
 
         database_configs.each do |db_config|
-          yield db_config.name
+          if db_config.database_tasks
+            yield db_config.name
+          else
+            next ### Skip registering and using this DB's tasks
```

Comment:
I'd prefer an early return here like this:

```ruby
database_configs.each do |db_config|
  next unless db_config.database_tasks

  yield db_config.name
end
```

Reply:
Updated.